src/arch/riscv/kernel/vspace.c

```diff
@@ -123,7 +122,7 @@ BOOT_CODE VISIBLE void map_kernel_window(void)
         pptr += RISCV_GET_LVL_PGSIZE(0);
         paddr += RISCV_GET_LVL_PGSIZE(0);
     }
-    /* now we should be mapping the 1GiB kernel base */
+    /* now map the 1GiB where kernel ELF image lives */
```

Comment:
Neither the old nor the new comment are great. Maybe instead: `/* now map KERNEL_ELF_BASE to KERNEL_ELF_PADDR_BASE as a 1 GiB window */`. Is the 1GiB actually correct for all configs? I haven't looked, but the could just below looks like it would depend on what size a level 0 page is.